+ Removed the wait in the parent as child process is not available.
+ Tests were failing to delete the temporary directory created for testing.
+ It is happening because after doing chroot on temp directory, the path will be changed and 
+ we cannot exit from chroot because child process is not available.
+ so created a temp directory for testing with fixed name which is being delted everytime before test runs.
diff --git a/testcases/kernel/syscalls/chroot/chroot02.c b/testcases/kernel/syscalls/chroot/chroot02.c
index e483ca4b5..11a7217f4 100644
--- a/testcases/kernel/syscalls/chroot/chroot02.c
+++ b/testcases/kernel/syscalls/chroot/chroot02.c
@@ -84,7 +84,7 @@ int main(int ac, char **av)
                if (pid == 0) {
                        retval = 0;

-                       if (chroot(tst_get_tmpdir()) == -1) {
+                       if (chroot("/tmp/chroot02_temp_dir") == -1) {
                                perror("chroot failed");
                                retval = 1;
                        } else {
@@ -94,14 +94,9 @@ int main(int ac, char **av)
                                }
                        }

-                       exit(retval);
                }

-               /* parent */
-               wait(&status);
-               /* make sure the child returned a good exit status */
-               if (WIFSIGNALED(status) ||
-                   (WIFEXITED(status) && WEXITSTATUS(status) != 0))
+               if (retval)
                        tst_resm(TFAIL, "chroot functionality incorrect");
                else
                        tst_resm(TPASS, "chroot functionality correct");
@@ -118,8 +113,10 @@ int main(int ac, char **av)
 void setup(void)
 {
        tst_require_root();
-
-       tst_tmpdir();
+
+       rmdir("/tmp/chroot02_temp_dir");
+       mkdir("/tmp/chroot02_temp_dir", 0644);
+       chdir("/tmp/chroot02_temp_dir");
        if ((fileHandle = creat(TMP_FILENAME, 0777)) == -1)
                tst_brkm(TBROK, cleanup, "failed to create temporary file "
                         TMP_FILENAME);
@@ -143,6 +140,4 @@ void cleanup(void)
         */
        close(fileHandle);

-       tst_rmdir();
-
 }


